axlearn/common/launch.py

```diff
@@ -70,7 +70,7 @@
 flags.DEFINE_string("jax_backend", None, "Specifies the XLA backend to use.", required=True)
 flags.DEFINE_string(
     "distributed_coordinator",
-    None,
+    os.environ.get("DISTRIBUTED_COORDINATOR", None),
```

Comment:
Do we need this change for the GPU runner, or can we just supply the flags for now?

Reply:
It gets ugly imo if you have to pass through these flags 2 times. You will end up having to pass it through as environment variables to the pod and then read those env variables within the container to pass it as flags.

Wouldn't it be nicer if we can pass env variables as default instead and only pass flags to override if needed?

The way it's implemented is backwards compatible, it will still return None as the default if the env flag is not set.

Reply:
For example, without pulling from env variables I would have to do this:
```
BASTION_TIER=1 axlearn gcp gke start --instance_type=gpu-a3-highgpu-8g-8 \
        --cluster=daily-run-single-zone-gker27 --num_replicas=1 \
        --queue multislice-queue \
        --bundler_type=artifactregistry --bundler_spec=image=gpu \
        --bundler_spec=dockerfile=Dockerfile --bundler_spec=target=gpu \
        -- mkdir -p /tmp/test_trainer\; python3 -m axlearn.common.launch_trainer_main \
  --module=text.gpt.c4_trainer --config=fuji-7B-v1-single-host \
  --trainer_dir=/tmp/test_trainer --data_dir=gs://axlearn-public/tensorflow_datasets \
  --jax_backend=gpu --distributed_coordinator=\$DISTRIBUTED_COORDINATOR \
  --process_id=\$PROCESS_ID ---num_processes=\$NUM_PROCESSES
  ```